src/main/java/org/ow2/proactive/connector/iaas/cloud/provider/azure/AzureProvider.java

```diff
@@ -0,0 +1,602 @@
+                                                                                                                                                                    .isPresent())
+                                                                                                                              .map(nicIpConfiguration -> nicIpConfiguration.getPublicIpAddress()
+                                                                                                                                                                           .ipAddress())
+                                                                                                                              .collect(Collectors.toList()))
+                                                                                                           .build())
+                                                    .status(String.valueOf(vm.powerState().toString()))
+                                                    .build())
+                                 .collect(Collectors.toSet());
+    }
+
+    @Override
+    public List<ScriptResult> executeScriptOnInstanceId(Infrastructure infrastructure, String instanceId,
+            InstanceScript instanceScript) {
+        VirtualMachine vm = azureProviderUtils.searchVirtualMachineByID(azureServiceCache.getService(infrastructure),
+                                                                        instanceId)
+                                              .orElseThrow(() -> new RuntimeException("ERROR unable to find instance with ID: '" +
+                                                                                      instanceId + "'"));
+        return executeScriptOnVM(vm, instanceScript);
+    }
+
+    @Override
+    public List<ScriptResult> executeScriptOnInstanceTag(Infrastructure infrastructure, String instanceTag,
+            InstanceScript instanceScript) {
+        VirtualMachine vm = azureProviderUtils.searchVirtualMachineByName(azureServiceCache.getService(infrastructure),
+                                                                          instanceTag)
+                                              .orElseThrow(() -> new RuntimeException("ERROR unable to find instance with name: '" +
+                                                                                      instanceTag + "'"));
+        return executeScriptOnVM(vm, instanceScript);
+    }
+
+    private List<ScriptResult> executeScriptOnVM(VirtualMachine vm, InstanceScript instanceScript) {
+
+        // Concatenate all provided scripts in one (Multiple VMExtensions per handler not supported)
+        StringBuilder concatenatedScripts = new StringBuilder();
+        Lists.newArrayList(instanceScript.getScripts()).forEach(script -> {
+            concatenatedScripts.append(script).append(';');
```

Comment:
Well, it seems that only single cmdline works, from the documentation it is not clear at all how the script is acutally executed:
https://github.com/Azure/custom-script-extension-linux.
All we can provide is a single String (without multi-line support) and after a few tests it seems that the script is executed within root user context (environment variables are sourced) on Linux.

Regarding OS type compatibility, they used to have a dedicated extension for Linux and another one for Windows (https://docs.microsoft.com/en-us/azure/virtual-machines/virtual-machines-windows-extensions-customscript). But it seems that the script extension feature is currently under intensive development and that they are trying to unify both.
As you can see the previous version (https://github.com/Azure/azure-linux-extensions/tree/master/CustomScript) used an explicit Linux-only name ('CustomScriptForLinux') while the new one is just "CustomScript".

I still changed the regular ';' separator by '&&' (from a static variable) which is compatible on Windows too.
Also, this is only required for InitScript (from `createInstance` method) as we can only give one extension. For the `executeScriptOnInstance` methods, they can be executed separately by updating the extension.
Overall, advanced scripts must be given from a file and imported into an Azure blob, as mentioned in the PR.
I'll will create a common 'script extension' issue after the merge including:
- multiple init scripts
- Windows compatibility
- advanced script upload